From 1fc185e7a509a778564a3a999cff7b8b74d3a299 Mon Sep 17 00:00:00 2001
From: Carlos Garnacho <carlosg@gnome.org>
Date: Mon, 7 Jul 2025 17:29:03 +0200
Subject: [PATCH 2/4] color: Use GAppLaunchContext to spawn color calibrator

In a worse situation than media-keys, we don't even attempt to integrate
with startup notification here, so using GdkAppLaunchContext is somewhat
pointless.
---
 plugins/color/gsd-color-calibrate.c | 30 ++++++++++-------------------
 plugins/color/main.c                |  5 -----
 plugins/color/meson.build           |  1 -
 3 files changed, 10 insertions(+), 26 deletions(-)

diff --git a/plugins/color/gsd-color-calibrate.c b/plugins/color/gsd-color-calibrate.c
index 6796925d..83beb673 100644
--- a/plugins/color/gsd-color-calibrate.c
+++ b/plugins/color/gsd-color-calibrate.c
@@ -23,7 +23,6 @@
 #include <glib/gi18n.h>
 #include <colord.h>
 #include <libnotify/notify.h>
-#include <gdk/gdk.h>
 
 #include "gsd-color-calibrate.h"
 
@@ -76,39 +75,30 @@ gcm_session_async_helper_free (GcmSessionAsyncHelper *helper)
 static void
 gcm_session_exec_control_center (GsdColorCalibrate *calibrate)
 {
-        gboolean ret;
-        GError *error = NULL;
-        GAppInfo *app_info;
-        GdkAppLaunchContext *launch_context;
+        g_autoptr (GError) error = NULL;
+        g_autoptr (GAppInfo) app_info = NULL;
+        g_autoptr (GAppLaunchContext) launch_context = NULL;
 
         /* setup the launch context so the startup notification is correct */
-        launch_context = gdk_display_get_app_launch_context (gdk_display_get_default ());
+        launch_context = g_app_launch_context_new ();
         app_info = g_app_info_create_from_commandline (BINDIR "/gnome-control-center color",
                                                        "gnome-control-center",
                                                        G_APP_INFO_CREATE_SUPPORTS_STARTUP_NOTIFICATION,
                                                        &error);
         if (app_info == NULL) {
                 g_warning ("failed to create application info: %s",
-                           error->message);
-                g_error_free (error);
-                goto out;
+                           error->message);\
+                return;
         }
 
         /* launch gnome-control-center */
-        ret = g_app_info_launch (app_info,
-                                 NULL,
-                                 G_APP_LAUNCH_CONTEXT (launch_context),
-                                 &error);
-        if (!ret) {
+        if (!g_app_info_launch (app_info,
+                                NULL,
+                                G_APP_LAUNCH_CONTEXT (launch_context),
+                                &error)) {
                 g_warning ("failed to launch gnome-control-center: %s",
                            error->message);
-                g_error_free (error);
-                goto out;
         }
-out:
-        g_object_unref (launch_context);
-        if (app_info != NULL)
-                g_object_unref (app_info);
 }
 
 static void
diff --git a/plugins/color/main.c b/plugins/color/main.c
index 1996a6d1..4d41bfea 100644
--- a/plugins/color/main.c
+++ b/plugins/color/main.c
@@ -1,13 +1,8 @@
-#include <gtk/gtk.h>
-
 #include "gsd-main-helper.h"
 #include "gsd-color-manager.h"
 
 int
 main (int argc, char **argv)
 {
-        gtk_init ();
-
         return gsd_main_helper (GSD_TYPE_COLOR_MANAGER, argc, argv);
 }
-
diff --git a/plugins/color/meson.build b/plugins/color/meson.build
index ed79538b..1c5be07f 100644
--- a/plugins/color/meson.build
+++ b/plugins/color/meson.build
@@ -14,7 +14,6 @@ deps = plugins_deps + [
   libgeoclue_dep,
   libnotify_dep,
   libcommon_dep,
-  gtk_dep,
   m_dep,
 ]
 
-- 
2.49.0

